File: common/src/main/java/me/reidj/forest/data/Knowledge.java
package me.reidj.forest.data;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author : Рейдж
 * @project : forest
 **/

@AllArgsConstructor
@Getter
public enum Knowledge {
    APPLE("apple", "Новый предмет"),
    FLINT("flint", "Новый предмет"),
    HEAL("heal", "Новый предмет"),
    TOXIC("red_mushroom", "Новый предмет"),
    STONE("stone", "Новый предмет"),
    COLD("cold", "Новая земля"),
    HOT("hot", "Новая земля"),
    ;

    private final String picture;
    private final String message;
}

File: forest-game/src/main/kotlin/me/reidj/forest/knowledge/Knowledge.kt
package me.reidj.forest.knowledge

import me.reidj.forest.user.User

enum class Knowledge(private val picture: String, private val message: String) {

    APPLE("apple", "Новый предмет"),
    FLINT("flint", "Новый предмет"),
    HEAL("heal", "Новый предмет"),
    TOXIC("red_mushroom", "Новый предмет"),
    STONE("stone", "Новый предмет"),
    COLD("cold", "Новая земля"),
    HOT("hot", "Новая земля"),;

    fun tryGive(user: User) {
        if (user.stat!!.knowledge.contains(this))
            return
        me.reidj.forest.channel.ModHelper.banner(user, picture, message)
        user.stat!!.knowledge.add(this)
    }
}